src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -142,9 +142,24 @@ private void LogSuccessfulTelemetryToClient(AuthenticationResult authenticationR
                 telemetryEventDetails.SetProperty(TelemetryConstants.PopToken, authenticationResult.TokenType.Equals(Constants.PoPTokenType));
                 telemetryEventDetails.SetProperty(TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds);
                 telemetryEventDetails.SetProperty(TelemetryConstants.ActivityId, authenticationResult.CorrelationId);
+                telemetryEventDetails.SetProperty(TelemetryConstants.RefreshOn, 
+                    authenticationResult.AuthenticationResultMetadata.RefreshOn.HasValue ?
+                    DateTimeHelpers.DateTimeToUnixTimestampMilliseconds(authenticationResult.AuthenticationResultMetadata.RefreshOn.Value)
+                    : 0);
+
+                LogDatapointsToTelemetryClient(telemetryDatapoints, telemetryEventDetails);
             }
         }
 
+        private void LogDatapointsToTelemetryClient(TelemetryDatapoints telemetryDatapoints, MsalTelemetryEventDetails telemetryEventDetails)
+        {
+            if (telemetryDatapoints == null) return;
```

Comment:
Ideally it will never be null as I instantiate this class in RequestContext so that we can capture latencies incrementally.

Reply:
So this check is not needed then.

Reply:
Do we log CacheUsed anywhere? Also in the scenario when static Msal is used (so MsalInternalCache only) we would always log 0 for L1Latency and 0 for DurationInCache. So from telemetry it would seem like cache is not used, but then DurationInHttp would also be 0 (since internal cache actually is used). I think we would have to measure and set L1Latency for internal cache if cache serialization is not enabled. (just brainstorming).

Reply:
Added cache used value instead based on discussions